include/deal.II/algorithms/theta_timestepping.h

```diff
@@ -236,10 +236,7 @@ namespace Algorithms
      * The current time in the timestepping scheme.
      */
     double current_time() const;
-    /**
-     * The current step size.
-     */
-    double step_size() const;
```

Comment:
Hm -- this is a public member function of the class. How do you know that it's not used anywhere?

Reply:
This method has no definition, so any attempt to use it would result in linker errors. The following test code fails for exactly that reason:


```cpp
#include <deal.II/lac/vector.h>
#include <deal.II/algorithms/theta_timestepping.h>
#include <deal.II/algorithms/theta_timestepping.templates.h>


using namespace dealii;

struct Test : public Algorithms::OperatorBase
{
  virtual void operator() (AnyData &, const AnyData &) override
  {
  }
};


int main()
{

  Test op;
  Algorithms::ThetaTimestepping<Vector<float>> timestepping(op, op);
  timestepping.step_size();
}
```

Reply:
This should be implemented though. Maybe open an issue?

Reply:
Apparently there has never been a use case, and unlike removing operators or constructors, not providing this function doesn't have any downsides. I'm generally of the opinion that we shouldn't implement things just because we can, rather than based on need.